internal/asm/f64/l2norm_noasm.go

```diff
@@ -34,3 +34,59 @@ func L2NormUnitary(x []float64) (sum float64) {
 	}
 	return scale * math.Sqrt(sumSquares)
 }
+
+// L2NormInc returns the L2-norm of x.
+func L2NormInc(x []float64, n, incX uintptr) (sum float64) {
```

Comment:
Is it necessary to name the return value? If it's for consistency with asm, then at least rename to `norm`? Similarly for the distance function.